docs/src/man/basics.md

```diff
@@ -0,0 +1,2042 @@
+2-element Vector{Any}:
+  "Mathematics"
+ 5
+
+julia> dfr
+DataFrameRow
+ Row │ marks  subject      semester
+     │ Int64  String       Int64
+─────┼──────────────────────────────
+   2 │    98  Mathematics         5
+
+julia> sdf = view(data, :, 2:3) # Column subsetting
+6×2 SubDataFrame
+ Row │ subject      semester
+     │ String       Int64
+─────┼───────────────────────
+   1 │ English             2
+   2 │ Mathematics         5
+   3 │ Economics           7
+   4 │ History             4
+   5 │ Economics           5
+   6 │ Psychology          6
+
+julia> sdf[2, :subject] = "Computer Science" # set value of col `:subject` in second row to `Computer Science` in-place
+"Computer Science"
+
+julia> sdf
+6×2 SubDataFrame
+ Row │ subject           semester
+     │ String            Int64
+─────┼────────────────────────────
+   1 │ English                  2
+   2 │ Computer Science         5
+   3 │ Economics                7
+   4 │ History                  4
+   5 │ Economics                5
+   6 │ Psychology               6
+
+julia> sdf[6, 1:2] = ["Biology", 7] # set value of multiple columns
+2-element Vector{Any}:
+  "Biology"
+ 7
+
+julia> sdf
+6×2 SubDataFrame
+ Row │ subject           semester
+     │ String            Int64
+─────┼────────────────────────────
+   1 │ English                  2
+   2 │ Computer Science         5
+   3 │ Economics                7
+   4 │ History                  4
+   5 │ Economics                5
+   6 │ Biology                  7
+
+julia> data[:, 3] = [4, 5, 7, 8, 2, 1]
+6-element Vector{Int64}:
+ 4
+ 5
+ 7
+ 8
+ 2
+ 1
+
+julia> data
+6×3 DataFrame
+ Row │ marks  subject           semester
+     │ Int64  String            Int64
+─────┼───────────────────────────────────
+   1 │    80  English                  4
+   2 │    98  Computer Science         5
+   3 │    78  Economics                7
+   4 │    95  History                  8
+   5 │    78  Economics                2
+   6 │    89  Biology                  1
+
+julia> data[!, :marks] .= [85, 89, 78, 58, 96, 68] # col `:marks` is replaced freshly allocated vector
+6-element Vector{Int64}:
+ 85
+ 89
+ 78
+ 58
+ 96
+ 68
+
+julia> data
+6×3 DataFrame
+ Row │ marks  subject           semester
+     │ Int64  String            Int64
+─────┼───────────────────────────────────
+   1 │    85  English                  4
+   2 │    89  Computer Science         5
+   3 │    78  Economics                7
+   4 │    58  History                  8
+   5 │    96  Economics                2
+   6 │    68  Biology                  1
+
+julia> data[!, :Students] .= ["Rohit", "Akshat", "Rahul", "Aayush", "Prateek", "Anam"] # allocates a new column `:Students` and adds it
+6-element Vector{String}:
+ "Rohit"
+ "Akshat"
+ "Rahul"
+ "Aayush"
+ "Prateek"
+ "Anam"
+
+julia> data[:, :City] .= ["Kanpur", "Lucknow", "Bhuvneshwar", "Jaipur", "Ranchi", "Dehradoon"] # allocates the column in-place because `:City` is not present in `data`
```

Comment:
allocation cannot happen in-place. Here `:` and `!` do the same. They would do a different thing for an existing column.

Reply:
Yeah, I have written a note on it after this example.